controllers/chat/getSignedChannelList.js

```diff
@@ -14,22 +22,36 @@ const __queryChannelBuilder = async (userId, limit, offset) => {
   return channels;
 };
 
-const __transformChannelData = (channel) => {
+/**
+ *
+ * @param {import('stream-chat').Channel} channel
+ * @returns
+ */
+const __filterAndTransformChannelData = (acc, channel) => {
+  console.log('channel');
+  console.log(channel);
   const newChannel = {...channel.data};
   delete newChannel.config;
   delete newChannel.own_capabilities;
 
+  const messageLength = channel.state.messages?.length;
+  const channelType = newChannel?.channel_type;
+
+  if (messageLength === 0 && channelType === CHANNEL_TYPE.TOPIC) return acc;
+
   const members = [];
   Object.keys(channel.state.members).forEach((member) => {
     members.push(channel?.state?.members[member]);
   });
 
-  return {
+  acc.push({
     ...newChannel,
     members,
     unreadCount: channel.state.unreadCount,
     messages: channel.state.messages
-  };
+  });
+
+  return acc;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The transformation of channel data has been modified to filter out channels with no messages and of type 'TOPIC'. This seems like a business logic decision and looks fine as long as it aligns with the application's requirements. However, the console logs on lines 31 and 32 should be removed or replaced with proper logging mechanisms to avoid exposing sensitive information in production environments.

```diff
- console.log('channel');
- console.log(channel);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->